posthog/hogql_queries/query_runner.py

```diff
@@ -68,14 +73,30 @@
 
 
 class ExecutionMode(IntEnum):
-    CALCULATION_ALWAYS = 2
+    CALCULATION_ALWAYS = 4
     """Always recalculate."""
-    RECENT_CACHE_CALCULATE_IF_STALE = 1
+    CALCULATE_ASYNC_ALWAYS = 3
+    """Always kick off async calculation."""
+    RECENT_CACHE_CALCULATE_BLOCKING_IF_STALE = 2
     """Use cache, unless the results are missing or stale."""
+    RECENT_CACHE_CALCULATE_ASYNC_IF_STALE = 1
+    """Use cache, kick off async calculation when results are missing or stale."""
     CACHE_ONLY_NEVER_CALCULATE = 0
     """Do not initiate calculation."""
 
 
+def execution_mode_from_refresh(refresh_requested: bool | str) -> ExecutionMode:
+    refresh_map = {
+        "stale": ExecutionMode.RECENT_CACHE_CALCULATE_BLOCKING_IF_STALE,
+        "async": ExecutionMode.RECENT_CACHE_CALCULATE_ASYNC_IF_STALE,
+        "force_async": ExecutionMode.CALCULATE_ASYNC_ALWAYS,
```

Comment:
I think this would be clearer at a glance:
```suggestion
        "if_stale": ExecutionMode.RECENT_CACHE_CALCULATE_BLOCKING_IF_STALE,
        "if_stale_async": ExecutionMode.RECENT_CACHE_CALCULATE_ASYNC_IF_STALE,
        "async": ExecutionMode.CALCULATE_ASYNC_ALWAYS,
```

Reply:
I don't really like the "if_stale" as it should be the underlying assumption that you get cached data if it's still reasonable. Here's my latest suggestion:

1.	"blocking":
Explanation: The term “blocking” conveys the idea of a synchronous or immediate calculation. It implies the method should use the cache if available and only perform a blocking (synchronous) calculation if the cache is stale. This key aligns with the default assumption of “if stale” behavior, meaning it will recalculate if the cached data is stale.
2.	"async":
Explanation: The term “async” indicates that the method should use the cache if available and perform an asynchronous calculation if the cache is stale. This follows the default assumption of recalculating “if stale” but does so asynchronously.
3.	"force_async":
Explanation: The prefix “force” indicates a deviation from the default behavior. “Force_async” means that the method should always initiate an asynchronous calculation, regardless of the cache’s state.
4.	"force_blocking":
Explanation: Similar to “force_async,” the prefix “force” signifies overriding the default behavior. “Force_blocking” means the method should always perform a blocking (synchronous) calculation, regardless of the cache’s state.
5.	"force_cache":
Explanation: The prefix “force” here indicates that the method should strictly use the cache and never initiate a calculation. This key explicitly directs the method to only rely on cached data and avoid any recalculations.